modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JSONUpgradeLiferayThemePackageJSONCheck.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.json.JSONObjectImpl;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.servlet.HttpMethods;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.version.Version;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * @author Nícolas Moura
+ */
+public class JSONUpgradeLiferayThemePackageJSONCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith("package.json") ||
+			!content.contains("\"liferayTheme\"")) {
+
+			return content;
+		}
+
+		JSONObject jsonObject = new JSONObjectImpl(content);
+
+		_formatDevDependencies(jsonObject);
+		_formatThemeVersion(jsonObject);
+		_formatScripts(jsonObject);
+
+		return JSONUtil.toString(jsonObject);
+	}
+
+	@Override
+	protected String[] getValidExtensions() {
+		return new String[] {"json"};
+	}
+
+	private void _formatDevDependencies(JSONObject jsonObject) {
+		Map<String, String> devDependenciesLatestVersions =
+			_getDevDependenciesLatestVersions();
+
+		if (devDependenciesLatestVersions.isEmpty()) {
+			return;
+		}
+
+		JSONObject devDependenciesJSONObject = jsonObject.getJSONObject(
+			"devDependencies");
+
+		if (devDependenciesJSONObject == null) {
+			return;
+		}
+
+		for (String packageName : _DEV_DEPENDENCIES_PACKAGES) {
+			String currentVersion = devDependenciesJSONObject.getString(
+				packageName);
+
+			if (currentVersion.equals("") &&
+				packageName.equals("liferay-font-awesome")) {
+
+				continue;
+			}
+
+			String latestVersion = devDependenciesLatestVersions.get(
+				packageName);
+
+			if (!currentVersion.equals(latestVersion)) {
+				devDependenciesJSONObject.put(packageName, latestVersion);
+
+				if (packageName.equals("liferay-theme-tasks")) {
+					devDependenciesJSONObject.put(
+						packageName, StringPool.CARET + latestVersion);
+				}
```

Comment:
@nicolas-sss, for the case of `liferay-theme-tasks`, we're performing 2 put operations, can we optimize?

I was thinking in something like:
```
if (packageName.equals("liferay-theme-tasks")) {
	devDependenciesJSONObject.put(
		packageName, StringPool.CARET + latestVersion);
}
else {
	devDependenciesJSONObject.put(packageName, latestVersion);
}
```